primitives/state-machine/src/overlayed_changes.rs

```diff
@@ -498,6 +510,44 @@ impl OverlayedChanges {
 		)
 	}
 
+	/// Get an iterator over all pending and committed child tries in the overlay.
+	pub fn child_infos(&self) -> impl IntoIterator<Item=&ChildInfo> {
+		self.prospective.children_default.iter()
+			.chain(self.committed.children_default.iter())
+			.map(|(_, v)| &v.1).collect::<BTreeSet<_>>()
```

Comment:
I like the idea of swapping them for consistency. However, I won't be optimizing it further with some iterator as this code is going away with storage tx when there is only one map.

Reply:
when will this go away? I am asking because when working on #3263 I got conflicting requirement (I did dumb down the code at some point but it introduces O(N), and https://github.com/paritytech/substrate/pull/3263#issuecomment-573559259 after I switch back to the old version (probably on of the reason the PR refactor did not reach consensus)).
Just not sure if it will be quick to put in place.

Reply:
```rust
	/// Get an iterator over all pending and committed child tries in the overlay.
	pub fn child_infos(&self) -> impl IntoIterator<Item=&ChildInfo> {
		use std::cmp::Ordering;
		let mut committed = self.committed.children_default.iter()
			.map(|(_, v)| &v.1);
		let mut prospective = self.prospective.children_default.iter()
			.map(|(_, v)| &v.1);
		let mut committed_next = committed.next();
		let mut prospective_next = prospective.next();
		std::iter::from_fn(move || {
			match committed_next.cmp(&prospective_next) {
				Ordering::Less => mem::replace(&mut committed_next, committed.next()),
				Ordering::Greater => mem::replace(&mut prospective_next, prospective.next()),
				Ordering::Equal => {
					committed_next = committed.next();
					mem::replace(&mut prospective_next, prospective.next())
				},
			}
		})
	}
```
or anything similar ?
but it is only a nitpick.

Reply:
Thing is that my change is not worsening the situation as it was building a `BTreeSet` before. When implementing something that is too complicated to verify that it is correct just by looking at it we need a test for that. With the absence of that I rather stick with the simple solution.

I appreciate your effort though.

Reply:
Np, I did not realize it was the existing code, actually this snippert that is probably some code I already did write many time in many of my unmerged pr(s).